tests/std/tests/P0896R4_ranges_alg_transform_binary/test.cpp

```diff
@@ -0,0 +1,154 @@
+
+// Validate dangling story
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<false>{}, borrowed<false>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<ranges::dangling, ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<false>{}, borrowed<true>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<ranges::dangling, int*, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<true>{}, borrowed<false>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<int*, ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<true>{}, borrowed<true>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<int*, int*, int*>>);
+
+struct instantiator {
+    static constexpr P input1[3]     = {{1, 99}, {4, 98}, {5, 97}};
+    static constexpr P input2[3]     = {{99, 6}, {98, 7}, {97, 8}};
+    static constexpr int expected[3] = {7, 11, 13};
+
+    static constexpr P shortInput1[2]     = {{1, 99}, {4, 98}};
+    static constexpr P shortInput2[2]     = {{99, 6}, {98, 7}};
+    static constexpr int shortExpected[2] = {7, 11};
+
+    template <ranges::input_range Read1, ranges::input_range Read2, weakly_incrementable Write>
+    static constexpr void call() {
+        using ranges::transform, ranges::binary_transform_result, ranges::iterator_t;
+        { // Validate iterator + sentinel overload
+            int output[3] = {-1, -1, -1};
+            Read1 wrapped_in1{input1};
+            Read2 wrapped_in2{input2};
+
+            auto result = transform(wrapped_in1.begin(), wrapped_in1.end(), wrapped_in2.begin(), wrapped_in2.end(),
+                Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out.peek() == output + 3);
+            assert(ranges::equal(output, expected));
+        }
+        { // Validate range overload
+            int output[3] = {-1, -1, -1};
+            Read1 wrapped_in1{input1};
+            Read2 wrapped_in2{input2};
+
+            auto result = transform(wrapped_in1, wrapped_in2, Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out.peek() == output + 3);
+            assert(ranges::equal(output, expected));
+        }
+        { // Validate range overload first range shorter

```

Comment:
We're missing coverage for the iterator overload with different length ranges. What do you think about dropping the equal-length test cases, and duplicating this different-length cases for the iterator overload?